app/components/project-geometries/modes/draw.js

```diff
@@ -97,7 +97,14 @@ export default class DrawComponent extends Component {
 
     // can't direct select a point
     if (selectedID && type !== 'Point' && mode === 'simple_select') {
-      draw.changeMode('direct_select', { featureId: selectedID });
+      const geometryType = this.get('geometryType');
+      if (geometryType === 'commercial-overlays'
+        || geometryType === 'special-purpose-districts'
+        || geometryType === 'underlying-zoning') {
+        draw.changeMode('direct_select_rezoning', { featureId: selectedID });
+      } else {
+        draw.changeMode('direct_select', { featureId: selectedID });
+      }
```

Comment:
Hmm we should really keep "geometry type" information out of this component as much as possible. Is there another way to approach this?

We talked, let's do this: instead of having these domain-specific details directly in draw, let's changed `geometryType` to `requireLabel` and make it a boolean, then change the specific templates for types (underlying-zoning, etc) to configure that boolean as follows:

```
  {{#component modeComponent requireLabel=true as |editMode|}}
    {{editMode.feature-label-form
        options=labelOptions}}
    {{editMode.annotations}}
    {{project-geometries/utils/canonical-geometry-reset
      model=model}}
  {{/component}}
```

This way, the "domain-specific templates" are "consuming" a more generic draw component. The lot selector might need this attr too, unfortunately, suggesting some problems with our setup BUT I think this is a little better. 

Happy to pitch in!!!

Reply:
We decided to have a default mode as a class attr, and make it configurable!